### PR TITLE
Add Javascript dependencies to Foreman Discovery spec

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

File: packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 Summary:    MaaS Discovery Plugin for Foreman
 Name:       %{?scl_prefix}rubygem-%{gem_name}
-Version:    %{mainver}
+Version:    16.3.1
```
Comment (lzap) [inline packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec:20]: The last time I did this Ewoudl asked me to keep this, not sure why tho. The script does this automatically...

File: packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec [hunk 2]
```diff
@@ -94,17 +103,27 @@ cp -pa .%{gem_dir}/* \
...
+%{foreman_webpack_foreman}
 
 %files doc
 %doc %{gem_docdir}
 %doc %{gem_instdir}/README.md
 %{gem_instdir}/test
 
 %changelog
+* Thu Oct 29 2020 Rahul Bajaj <rahulrb0509@gmail.com> 16.3.1-1
+- Update to 16.3.1
```
Comment (lzap) [inline packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec:125]: Please add a note about adding webpack stuff.

File: packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec [hunk 3]
```diff
@@ -94,17 +103,27 @@ cp -pa .%{gem_dir}/* \
 %{gem_libdir}
 %{gem_instdir}/locale
 %exclude %{gem_cache}
+%exclude %{gem_instdir}/.eslintrc
+%exclude %{gem_instdir}/.stylelintrc
```
Comment (lzap) [inline packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec:107]: These RC files are not part of the gem, you can drop those excludes. They will probably fail the build.

File: packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec (hunk 1 above)
Comment (rabajaj0509) [reply packages/plugins/rubygem-foreman_discovery/rubygem-foreman_discovery.spec:20]: If we keep it as `%{mainver}` it fails at the `Lint Spec` stage in Jenkins.

**Comment by rabajaj0509** [comment]
I think the error boils down to:
```
Processing files: rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch
error: File not found: /builddir/build/BUILDROOT/rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch/usr/share/gems/gems/foreman_discovery-16.3.1/public/assets/foreman_discovery
error: File not found: /builddir/build/BUILDROOT/rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch/usr/share/gems/gems/foreman_discovery-16.3.1/public/webpack/foreman_discovery
error: File not found: /builddir/build/BUILDROOT/rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch/usr/share/foreman/public/webpack/foreman_discovery
RPM build errors:
    File not found: /builddir/build/BUILDROOT/rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch/usr/share/gems/gems/foreman_discovery-16.3.1/public/assets/foreman_discovery
    File not found: /builddir/build/BUILDROOT/rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch/usr/share/gems/gems/foreman_discovery-16.3.1/public/webpack/foreman_discovery
    File not found: /builddir/build/BUILDROOT/rubygem-foreman_discovery-16.3.1-1.fm2_3.el8.noarch/usr/share/foreman/public/webpack/foreman_discovery
```

**Comment by rabajaj0509** [comment]
I think this PR is ready now :+1: @lzap @evgeni

**Review by tbrisker — APPROVED** [review]

**Review by evgeni — APPROVED** [review]

**Comment by lzap** [comment]
@rabajaj0509 could you test the javascript stuff for the nightly build tomorrow please once 16.3.1 appears in http://yum.theforeman.org/plugins/nightly/el7/x86_64/ to ensure all artifacts are in place? Thanks for taking care!

**Pull request merged**